### DISCUSSION TITLE
[Share] Copy Citation Keys to Clipboard

### DISCUSSION THREAD

**Discussion by rodrigolourencofarinha** [windingwind/zotero-actions-tags#396] [Action Scripts]
### Description

This script selects citation keys from the currently selected Zotero items, adds an `@` prefix to each citation key, and copies the formatted citation keys to the clipboard, allowing for easy reference in documents.

### Event

None

### Operation

Script

### Data

```javascript
// Copy Citation Keys with @ prefix
// Based on copy DOIs

var citationKeyArray = [];
var items = Zotero.getActiveZoteroPane().getSelectedItems();
for (let item of items) {
    if (item.isRegularItem() && !(item instanceof Zotero.Collection)) {
        let citationKey = item.getField('citationKey'); // Replace 'citationKey' with the correct method if needed
        if (citationKey && !citationKeyArray.includes(citationKey)) {
            citationKeyArray.push('@' + citationKey); // Add @ prefix to the citation key
        }
    }
}
var clipboardText = citationKeyArray.join('\r\n');
Zotero.Utilities.Internal.copyTextToClipboard(clipboardText);
return "Citation Keys copied";
```

**Comment by windingwind** [comment]
https://github.com/windingwind/zotero-actions-tags?tab=readme-ov-file#custom-script

Please always use the items instead of ZoteroPane.getSelectedItems(). The action can be triggered from entrances outside the library, where the items are not from ZoteroPane.

**Reply by drlaru** [reply]
could you be more specific about the correction? I'm trying to use this and don't understand what needs to be fixed.
